src/cgutils.cpp

```diff
@@ -1651,11 +1655,15 @@ static jl_cgval_t typed_store(jl_codectx_t &ctx,
         bool maybe_null_if_boxed, const jl_cgval_t *modifyop, const std::string &fname)
 {
     auto newval = [&](const jl_cgval_t &lhs) {
-        const jl_cgval_t argv[3] = { cmp, lhs, rhs };
+        jl_cgval_t argv[3] = { cmp, lhs, rhs };
         jl_cgval_t ret(ctx.builder.getContext());
-        if (modifyop) {
-            ret = emit_invoke(ctx, *modifyop, argv, 3, (jl_value_t*)jl_any_type);
-        }
+        if (modifyop)
+            if (jl_typeis(modifyop->constant, jl_method_instance_type))
+                ret = emit_invoke(ctx, *modifyop, argv, 3, (jl_value_t*)jl_any_type);
+            else {
+                assert(jl_typeis(modifyop->constant, jl_intrinsic_type));
+                ret = emit_intrinsic(ctx, modifyop->constant, &argv[1], 2);
```

Comment:
Flag `intrinsic_rmw` within here?

Reply:
This is inside an anonymous function `newval(lhs)` called after the point at which `intrinsic_rmw` is set. So, it is too late to set it here. Or do you mean something else?

Reply:
Ops, I actually missed this is inside an anonymous function. Never mind!

Reply:
Seems like inference/codegen made a severe mistake if codegen needs to do dispatch now to check the type of the special operand? Shouldn't this be a check of the expression `!modifyop && cmp.constant && jl_typeis(cmp.constant, jl_intrinsic_type)`?